pkg/loki/filter.go

```diff
@@ -172,13 +187,82 @@ func notContainsKeyLineFilter(key string) lineFilter {
 		key:       key,
 		strictKey: true,
 		not:       true,
+		moreThan:  false,
+	}
+}
+
+func moreThanRegex(sb *strings.Builder, value string) {
+	// match each number greater than specified value using regex
+	// example for 123:
+	// ^ ( 12[3-9] | 1[3-9][0-9]+ | [2-9][0-9]+ | [1-9][0-9]{3,} )
+	//       |            |              |             |
+	//       |            |              |              ↪ match number more than 1000
+	//       |            |              |
+	//       |            |               ↪ match number from 200 to 999
+	//       |            |
+	//       |             ↪ match numbers from 130 to 199
+	//       |
+	//        ↪ match any number from 123 to 129
+
+	sb.WriteString("^(")
```

Comment:
I don't get correct results with `^` here , it's not the start of the line is it? I tried without carat and matches well: https://regex101.com/r/l8OWLf/1